test/bun.js/serve.test.ts

```diff
@@ -5,161 +5,145 @@ import { resolve } from "path";
-          port: port++,
-          development: false,
-          error(e) {
-            pass = true;
-            return new Response("fail", { status: 500 });
-          },
-
-          fetch(req) {
-            return new Response(
-              new ReadableStream({
-                pull(controller) {
-                  throw new Error("error");
-                },
-              }),
-            );
-          },
-        });
-
+      var pass = false;
+      await runTest({
+        development: false,
+        error(e) {
+          pass = true;
+          return new Response("PASS", { status: 555 });
+        },
+        fetch(req) {
+          return new Response(
+            new ReadableStream({
+              pull(controller) {
+                throw new Error("FAIL");
+              },
+            }),
+          );
+        },
+      }, async (server) => {
         const response = await fetch(
           `http://${server.hostname}:${server.port}`,
         );
         if (response.status > 0) {
-          expect(response.status).toBe(500);
-          expect(await response.text()).toBe("fail");
+          expect(response.status).toBe(555);
+          expect(await response.text()).toBe("PASS");
         }
-
         expect(pass).toBe(true);
-      } catch (e) {
-        throw e;
-      } finally {
-        server?.stop();
-      }
+      });
     });
 
     it("throw on pull after writing should not call the error handler", async () => {
-      var server;
-      try {
-        var pass = true;
-        server = serve({
-          port: port++,
-          development: false,
-          error(e) {
-            pass = false;
-            server?.stop();
-            server = null;
-            return new Response("fail", { status: 500 });
-          },
-
-          fetch(req) {
-            return new Response(
-              new ReadableStream({
-                pull(controller) {
-                  controller.enqueue("such fail");
-                  throw new Error("error");
-                },
-              }),
-            );
-          },
-        });
-
+      var pass = true;
+      await runTest({
+        development: false,
+        error(e) {
+          pass = true;
+          return new Response("FAIL", { status: 555 });
+        },
+        fetch(req) {
+          return new Response(
+            new ReadableStream({
+              pull(controller) {
+                controller.enqueue("PASS");
+                throw new Error("error");
+              },
+            }),
+          );
+        },
+      }, async (server) => {
         const response = await fetch(
           `http://${server.hostname}:${server.port}`,
         );
         // connection terminated
-        if (response.status > 0) {
-          expect(response.status).toBe(200);
-          expect(await response.text()).toBe("such fail");
-        }
+        expect(response.status).toBe(500);
+        expect(await response.text()).toBe("PASS");
```

Comment:
This test fails to begin with &minus; so I just fix up the expected results based on description, which remains to fail but at least it's a good starting point.